gem/lib/mulang/inspection.rb

```diff
@@ -0,0 +1,63 @@
+module Mulang
+  class Inspection
+    module Compacted
+      def as_json(*args)
+        super(*args).compact
+      end
+    end
+
+    include Mulang::Inspection::Compacted
+
+    REGEXP = Regexp.new %q{
+      ^(?<negation>Not\:)?
```

Comment:
I am looking for recommendations from @julian-berbel

Reply:
Looks pretty good! I'd just extract that repeated type group capture and remove unneeded backslashes:
```ruby
REGEXP = Regexp.new %q{
      ^(?<negation>Not:)?
      (?<type>[^:]+)
      (
        |
        :(?<matcher>WithLiteral|WithNonliteral|WithLogic|WithMath|WithFalse|WithNil|WithTrue) |
        :(?<matcher>WithChar|WithNumber|WithString|WithSymbol):(?<value>[^:]+) |
        :(?<target>[^:]+)(:(?<matcher>[^:]+)(:(?<value>[^:]+))?)?
      )$}.gsub(/\s/, '')
```